core/store/src/test_utils.rs

```diff
@@ -15,3 +19,49 @@ pub fn create_trie() -> Arc<Trie> {
     let store = create_test_store();
     Arc::new(Trie::new(store))
 }
+
+pub fn gen_changes(rng: &mut impl Rng, max_size: usize) -> Vec<(Vec<u8>, Option<Vec<u8>>)> {
+    let alphabet = &b"abcdefgh"[0..rng.gen_range(2, 8)];
+    let max_length = rng.gen_range(2, 8);
+
+    let mut state: HashMap<Vec<u8>, Vec<u8>> = HashMap::new();
+    let mut result = Vec::new();
+    let delete_probability = rng.gen_range(0.1, 0.5);
+    let size = rng.gen_range(0, max_size) + 1;
+    for _ in 0..size {
+        let key_length = rng.gen_range(1, max_length);
+        let key: Vec<u8> = (0..key_length).map(|_| alphabet.choose(rng).unwrap().clone()).collect();
+
+        let delete = rng.gen_range(0.0, 1.0) < delete_probability;
+        if delete {
+            let mut keys: Vec<_> = state.keys().cloned().collect();
+            keys.push(key);
+            let key = keys.choose(rng).unwrap().clone();
+            state.remove(&key);
+            result.push((key.clone(), None));
+        } else {
+            let value_length = rng.gen_range(1, max_length);
+            let value: Vec<u8> =
+                (0..value_length).map(|_| alphabet.choose(rng).unwrap().clone()).collect();
+            result.push((key.clone(), Some(value.clone())));
+            state.insert(key, value);
+        }
+    }
+    result
+}
+
+pub fn simplify_changes(
+    changes: &Vec<(Vec<u8>, Option<Vec<u8>>)>,
```

Comment:
this function discards deletions - we should change it or move back to private

Reply:
Could you please discuss about it with @frol and decide together? He's explanation why to move it here was good enough to me. I'm okay with any solution.

Reply:
My reasoning was to move it from `tests` since `trie/trie_tests.rs` was trying to import it, and that was completely unexpected to me. We can make it `pub(crate)`

Reply:
Is it enough for all of us to change `pub fn simplify_changes` to `#[allow(unused)]
pub(crate) fn simplify_changes`?

Reply:
`#[cfg(test)] pub(crate)` sounds better to me